README.future.md

```diff
@@ -2,99 +2,563 @@
+{
+  name: 'collection3',
+  size: 3126700,
+  status: 'Ready',
+  dimension: 3,
+  vectorCount: 99
+}
 ```
 
-### Configure an index
+### List Collections
 
-You can adjust the number of replicas or scale to a larger pod size (specified with `podType`). See [Pod types and sizes](https://docs.pinecone.io/docs/indexes#pods-pod-types-and-pod-sizes). You cannot downgrade pod size or change the base pod type.
+The result of this command is an array of collection names.
 
+```typescript
+> const list = await pinecone.listCollections();
+["collection1", "collection2"]
 ```
-await client.configureIndex('my-index', { replicas: 3, podType: 'p1.x2' })
-const config = await client.describeIndex('my-index')
-// {
-//   database: {
-//     name: 'my-index',
-//     metric: 'cosine',
-//     pods: 3,
-//     replicas: 3,
-//     shards: 1,
-//     podType: 'p1.x2'
-//   },
-//   status: { ready: false, state: 'ScalingUpPodType' }
-// }
+
+## Index operations
+
+Pinecone indexes support operations for working with vector data using operations such as upsert, query, fetch, and delete.
+
+### Targeting an index
+
+To perform data operations on an index, you target it using the `index` method.
+
+```typescript
+const pinecone = new Pinecone();
+const index = pinecone.index('test-index');
+
+// Now perform index operations
+await index.fetch(['1']);
+```
+
+### Targeting a namespace
+
+> [!NOTE]
+> Indexes in the [gcp-starter environment](https://docs.pinecone.io/docs/starter-environment) do not support namespaces.
+
+By default all data operations take place inside the default namespace of `''`. If you are working with other non-default namespaces, you can target the namespace by chaining a call to `namespace()`.
+
+```typescript
+const pinecone = new Pinecone();
+const index = pinecone.index('test-index').namespace('ns1');
+
+// Now perform index operations in the targeted index and namesapce
+await index.fetch(['1']);
+```
+
+If needed, you can check the currently targeted index and namespace by inspecting the `target` property of an index object.
+
+```typescript
+const pinecone = new Pinecone();
+const index = pinecone.index('test-index').namespace('ns1');
+
+console.log(index.target); // { index: 'test-index', namespace: 'ns1' }
 ```
+
+See [Using namespaces](https://docs.pinecone.io/docs/namespaces) for more information.
+
+### Upsert vectors
+
+Pinecone expects vector data to have the following form:
+
+```typescript
+type Vector = {
+  id: string;
+  values: Array<number>;
+  sparseValues?: Array<number>;
+  metadata?: object;
+};
+```
+
+To upsert some vectors, you can use the client like so.
+
+```typescript
+const pinecone = new Pinecone();
+
+// Target an index
+const index = pinecone.index('sample-index');
+
+// Prepare your data
+const vectors = [
+  {
+    id: '1',
+    values: [0.236, 0.971, 0.559],
+  },
+  {
+    id: '2',
+    values: [],
+  },
+];
+
+// Upsert the data into your index
+await index.upsert(vectors);
+```
+
```

Comment:
**TODO / non-blocking:** Add a section for using `batchSize` along with `upsert` when that PR lands.

I've seen somewhere that the maximum number of vectors per upsert call is 1000, but we only mention a "batch limit" in docs of 100 vectors: https://docs.pinecone.io/reference/upsert

 We could probably clarify some of that here when discussing batching, if that makes sense.